src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionMatcher.hpp

```diff
@@ -184,11 +189,11 @@ class VersionMatcher final
      * @param versionA string version item A to compare
      * @param versionB string version item B to compare
      * @param type VersionObjectType of the version strings A and B to compare.
-     * @return int
+     * @return VersionComparisonResult result of the comparison.
      */
-    static int compare(const std::string& versionA,
-                       const std::string& versionB,
-                       VersionObjectType type = VersionObjectType::Unspecified)
+    static VersionComparisonResult compare(const std::string& versionA,
```

Comment:
With the change in this function, tests show an anomaly behavior, but I think could be easily fixed.

Currently, the RPM versions and DPKG, have the following return statements:

https://github.com/wazuh/wazuh/blob/71a5e04bbf6508c7cdc05c65af535a057a5a6998/src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectRpm.hpp#L333

https://github.com/wazuh/wazuh/blob/71a5e04bbf6508c7cdc05c65af535a057a5a6998/src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectDpkg.hpp#L240

I thought and tested a simple condition:

```cpp
return output. version.size() > 0 && output.revision.size() > 0;
```

With this fix, the scanner shows:

```text
wazuh-modulesd:vulnerability-scanner:packageScanner.hpp:125 operator() : Scanning package - 'systemd' (Installed Version: %%%%%%%%%, Security Vulnerability: CVE-2023-7008). Identified vulnerability: Version: 0. Required Version Threshold: . Required Version Threshold (or Equal): 252.19-1~deb12u1.
wazuh-modulesd:vulnerability-scanner:versionMatcher.hpp:126 createVersionObject : Error creating VersionObject (DPKG). Version string doesn't match the specified type. Version string: %%%%%%%%%
wazuh-modulesd:vulnerability-scanner:versionMatcher.hpp:126 createVersionObject : Error creating VersionObject (DPKG). Version string doesn't match the specified type. Version string: 0
wazuh-modulesd:vulnerability-scanner:packageScanner.hpp:276 operator() : Failed to scan package: 'systemd', CVE Numbering Authorities (CNA): 'debian', Error: 'Unable to compare versions (%%%%%%%%% vs 0).'
wazuh-modulesd:vulnerability-scanner:packageScanner.hpp:328 handleRequest : Vulnerability scan for package 'systemd' on Agent '001' has completed.
```

But the `0` version also shows an error, should be in the content type of the version?

Reply:
Added handling for this scenarios